### PR TITLE
WT-7437 Upgrade documentation to doxygen 1.8.17

### PR DESCRIPTION
**Pull request by keithbostic**
@lukech, @jiechenbo: e1adc17 and 06b1413 upgrade the docs to doxygen 1.8.17 and skips building the documentation if the installed doxygen isn't that version or newer. (That means the documentation won't build until we build on a system with 1.8.17 installed, of course.)

@ddanderson, 65d64c8 and 298b964 are yours. I ended up leaving the architecture files in the `src/docs` directory, as that seemed simplest. I labeled them with a "warning: maybe not correct" message, and de-emphasized them in the introductory documentation page.

### CODE REVIEW

File: src/docs/Doxyfile [hunk 1]
```diff
@@ -113,607 +139,759 @@ ABBREVIATE_BRIEF       = "The $name class" \
...
-# tags, which will be replaced by the file and line number from which the
-# warning originated and the warning text. Optionally the format may contain
-# $version, which will be replaced by the version of the file (if it could
-# be obtained via FILE_VERSION_FILTER)
+# If the WARN_AS_ERROR tag is set to YES then doxygen will immediately stop when
+# a warning is encountered.
+# The default value is: NO.
+
+WARN_AS_ERROR          = NO
+
```
Comment (ddanderson) [inline src/docs/Doxyfile:841]: I don't know if we currently have any warnings when building, perhaps this could be `YES`?

**Review by ddanderson — APPROVED** [review]
LGTM.  I flagged an option `WARN_AS_ERROR` that we might consider having on.

File: src/docs/Doxyfile (hunk 1 above)
Comment (keithbostic) [reply src/docs/Doxyfile]: We currently build cleanly, I've turned the option on.

**Review by jiechenbo — APPROVED** [review]
LGTM!

**Comment by lukech** [comment]
@keithbostic Our `doc-update` Evergreen task is currently running on Ubuntu 18.04 distro (Doxygen 1.8.13). With the current changes in this PR the doc build step would be skipped/failed. I think we need to upgrade the distro of the task to Ubuntu 20.04 (Doxygen 1.8.17) to keep the task working - actually I created a patch build with the new distro update but ran into some Doxygen issues (please refer to the Jira ticket for details). Can you please take a look?

**Comment by keithbostic** [comment]
We're not going to merge this PR, deleting it.

**Review by lukech — APPROVED** [review]
LGTM

**Pull request merged**